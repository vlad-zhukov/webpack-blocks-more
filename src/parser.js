/*
 * Parser options.
 */
export default function parser(options) {
    return (context, {addLoader}) =>
        addLoader({
            parser: options,
        });
}
