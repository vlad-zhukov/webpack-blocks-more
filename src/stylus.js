/*
 * Stylus loader.
 */
export default function stylus(options = {}) {
    const {
        fileOptions = {}, extractOptions = {}, cssOptions = {}, stylusOptions = {},
    } = options;

    return (context, {addLoader}) =>
        addLoader({
            ...context.match,
            use: [
                {
                    loader: 'file-loader',
                    options: fileOptions,
                },
                {
                    loader: 'extract-loader',
                    options: extractOptions,
                },
                {
                    loader: 'css-loader',
                    options: cssOptions,
                },
                {
                    loader: 'stylus-loader',
                    options: Object.assign(
                        {
                            compress: true,
                        },
                        stylusOptions
                    ),
                },
            ],
        });
}
