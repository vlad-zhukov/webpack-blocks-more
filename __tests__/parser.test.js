import {createConfig, match} from '@webpack-blocks/webpack';
import parser from '../src/parser';
import stylus from '../src/stylus';

it('should add a parser rule', () => {
    const config = createConfig([
        parser({
            amd: false,
            browserify: false,
            requireJs: false,
            system: false,
            requireInclude: false,
            requireEnsure: false,
        }),
    ]);

    expect(config).toMatchSnapshot();
});

it('should work with other rules', () => {
    const config = createConfig([
        parser({
            amd: false,
            browserify: false,
            requireJs: false,
            system: false,
            requireInclude: false,
            requireEnsure: false,
        }),
        match('*.styl', [stylus()]),
    ]);

    expect(config).toMatchSnapshot();
});
