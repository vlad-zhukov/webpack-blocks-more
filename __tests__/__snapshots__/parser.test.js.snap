// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a parser rule 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "parser": Object {
          "amd": false,
          "browserify": false,
          "requireEnsure": false,
          "requireInclude": false,
          "requireJs": false,
          "system": false,
        },
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;

exports[`should work with other rules 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "parser": Object {
          "amd": false,
          "browserify": false,
          "requireEnsure": false,
          "requireInclude": false,
          "requireJs": false,
          "system": false,
        },
      },
      Object {
        "test": /\\^\\.\\*\\\\\\.styl\\$/,
        "use": Array [
          Object {
            "loader": "extract-loader",
            "options": Object {},
          },
          Object {
            "loader": "css-loader",
            "options": Object {},
          },
          Object {
            "loader": "stylus-loader",
            "options": Object {
              "compress": true,
              "disable-cache": true,
              "preferPathResolver": "webpack",
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;
